src/librustc_codegen_utils/codegen_backend.rs

```diff
@@ -136,24 +135,15 @@ impl CodegenBackend for MetadataOnlyCodegenBackend {
         ::symbol_names_test::report_symbol_names(tcx);
         ::rustc_incremental::assert_dep_graph(tcx);
         ::rustc_incremental::assert_module_sources::assert_module_sources(tcx);
-        ::rustc_mir::monomorphize::assert_symbols_are_distinct(tcx,
```

Comment:
Note to anybody looking through this: `collect_and_partition_mono_items` query does this assert internally.